graph_model/trie_model/trie.py

```diff
@@ -0,0 +1,42 @@
+import json
+from typing import TextIO
+from .trie_node import TrieNode
+class Trie:
+    def __init__(self):
+        self.root = TrieNode()
+
+    def add_data(self, new_suggestions):
+        for suggestion in new_suggestions:
+            self.root.insert(suggestion)
+
+    
+    def autocomplete(self, prefix, top_n):
+        cur_node = self.root.find_node(prefix)
+        
+        if len(cur_node.children) < 1:
+            return []
+
+        result = []
+        for i in cur_node.get_all_titles():
+            result.append(i)
+
+        # sort output titles by score
+        result.sort(key=lambda item: item['score'], reverse=True)
+        return result[:top_n]
+
+
+    def save(self, fileobject: TextIO):
+        self.root = self.root.to_dict()
+        json.dump(self.__dict__, fileobject)
+
+
+    def load(self, fileobject: TextIO):
+        def init_trie_node(node_dict):
```

Comment:
Usually it's not a good idea to define functions on runtime, please put it into class methods. Prepend the function name with underscore to suggest it's private.